singularity-eos/eos/modifiers/zsplit_eos.hpp

```diff
@@ -0,0 +1,274 @@
+    return scale * et;
+  }
+
+  template <typename Indexer_t = Real *>
+  PORTABLE_FUNCTION Real PressureFromDensityTemperature(
+      const Real rho, const Real temperature, Indexer_t &&lambda = nullptr) const {
+    const Real scale = GetScale_(lambda);
+    const Real Pt = t_.PressureFromDensityTemperature(rho, temperature, lambda);
+    return scale * Pt;
+  }
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real PressureFromDensityInternalEnergy(
+      const Real rho, const Real sie,
+      Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const Real iscale = GetInvScale_(lambda);
+    const Real scale = GetScale_(lambda);
+    const Real Pt = t_.PressureFromDensityInternalEnergy(rho, iscale * sie, lambda);
+    return scale * Pt;
+  }
+
+  template <typename Indexer_t = Real *>
+  PORTABLE_FUNCTION Real
+  MinInternalEnergyFromDensity(const Real rho, Indexer_t &&lambda = nullptr) const {
+    const Real scale = GetScale_(lambda);
+    const Real et = t_.MinInternalEnergyFromDensity(rho, lambda);
+    return scale * et;
+  }
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real
+  EntropyFromDensityTemperature(const Real rho, const Real temperature,
+                                Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const Real scale = GetScale_(lambda);
+    return scale * t_.EntropyFromDensityTemperature(rho, temperature, lambda);
+  }
+  template <typename Indexer_t = Real *>
+  PORTABLE_FUNCTION Real EntropyFromDensityInternalEnergy(
+      const Real rho, const Real sie, Indexer_t &&lambda = nullptr) const {
+    const Real scale = GetScale_(lambda);
+    const Real iscale = GetInvScale_(lambda);
+    return scale * t_.EntropyFromDensityInternalEnergy(rho, iscale * sie, lambda);
+  }
+
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real SpecificHeatFromDensityTemperature(
+      const Real rho, const Real temperature,
+      Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const Real scale = GetScale_(lambda);
+    return scale * t_.SpecificHeatFromDensityTemperature(rho, temperature, lambda);
+  }
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real SpecificHeatFromDensityInternalEnergy(
+      const Real rho, const Real sie,
+      Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const Real scale = GetScale_(lambda);
+    const Real iscale = GetInvScale_(lambda);
+    return scale * t_.SpecificHeatFromDensityInternalEnergy(rho, iscale * sie, lambda);
+  }
+
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real BulkModulusFromDensityTemperature(
+      const Real rho, const Real temperature,
+      Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const Real scale = GetScale_(lambda);
+    return scale * t_.BulkModulusFromDensityTemperature(rho, temperature, lambda);
```

Comment:
I _think_ this is a better approximation than for the heat capacity. I haven't looked at ionization models in great detail, but I'd initially imagine that the ionization state is a weak function of state along an isentrope.

In fact, we can estimate its importance by assuming that ionization is a pure function of temperature and ask how the temperature changes along an isentrope. Using some thermodynamic relations, the temperature changes along an isentrope as
```math
\left(\frac{\partial T}{\partial V}\right)_S = -\frac{T \alpha B_T}{C_V} = \Gamma \frac{T}{V}
```
For an ideal gas, this can be simplified to $P / C_V$, which is actually a stronger than I thought it would be. Without an explicit ionization model of course, this is impossible to take into account.

Regardless, it might be worth including a statement about the fact that the derivatives in these models do not take into account changing ionization state, which might be important in some applications.

Reply:
:+1: Clarified in the documentation with a warning and added a comment inline in the code.

After we start thinking more carefully about ionization models, I wonder if we could include this information inline in the sense that we could maybe make zsplit depend on an ionization model with a template parameter or something similar. Food for thought down the road.